### PR TITLE
Add `ignore_keys` argument to `convert_all_strings_to_quantity`

### PR DESCRIPTION
**Pull request by mattwthompson**
This PR implements @lilyminium's suggested fix to #1493 and also adds a test to catch some of the behavior she observed, which was previously not covered in tests.

- [x] Tag issue being addressed
- [x] Add [tests](https://github.com/openforcefield/openff-toolkit/tree/main/openff/toolkit/tests)
- [ ] Update docstrings/[documentation](https://github.com/openforcefield/openff-toolkit/tree/main/docs), if applicable
- [x] [Lint](https://open-forcefield-toolkit.readthedocs.io/en/latest/developing.html#style-guide) codebase
- [ ] Update [changelog](https://github.com/openforcefield/openff-toolkit/blob/main/docs/releasehistory.rst)

### CODE REVIEW

File: openff/toolkit/tests/test_forcefield.py [hunk 1]
```diff
@@ -998,6 +998,25 @@ def test_create_forcefield_from_xml_string(self):
...
+            parameter_handler = sage.get_parameter_handler(parameter_handler_name)
+
+            for parameter in parameter_handler.parameters:
+                assert isinstance(parameter.smirks, str)
+                assert not isinstance(parameter.smirks, unit.Quantity)
+
+                if (
+                    parameter_handler_name == "LibraryCharges"
+                    and getattr(parameter, "name") is not None
+                ):
```
Comment (j-wags) [inline openff/toolkit/tests/test_forcefield.py:1016]:
```suggestion
                # Ensure that, for example, F isn't converted to Farad 
                if (
                    parameter_handler_name == "LibraryCharges"
                    and getattr(parameter, "name") is not None
                ):
```

**Review by j-wags — APPROVED** [review]
Thanks for working with my extremely janky serialization code. Hopefully openff-models will enable us to make the migration to a cleaner, safer pydantic-based FF class.

**Comment by mattwthompson** [comment]
Thanks! And it's _our_ (de)serialization code at this point 😁

I'd like @lilyminium to have a look, particularly if `ignore_keys=["smirks", "name"],` is the right way to call this

**Comment by lilyminium** [comment]
Thanks @mattwthompson, that was a super fast fix! A more long-term solution might be to work with `ParameterAttribute.unit` somehow (i.e. not convert if there is no unit attached), but that looks like it'll take more work and this is an effective patch. LGTM!

With regards to additional keywords to ignore -- `id` might eventually become problematic if numbers aren't attached, but really `smirks` was the one breaking everything and iirc we don't guarantee anything about the `name` and `id` attributes.

**Comment by mattwthompson** [comment]
Thanks!

**Pull request merged**